src/cq.rs

```diff
@@ -33,6 +41,65 @@ impl CompletionQueueHandle {
     pub fn new() -> CompletionQueueHandle {
         CompletionQueueHandle {
             cq: unsafe { grpc_sys::grpc_completion_queue_create_for_next(ptr::null_mut()) },
+            ref_cnt: AtomicIsize::new(1),
+        }
+    }
+
+    fn add_ref(&self) -> Result<()> {
+        loop {
```

Comment:
why not using SpinLock to simplify the code?

Reply:
I don't think use `SpinLock` is appropriate. Just like we won't use `SpinLock<T>` to replace `AtomicT`, because `AtomicT` is exactly what we want.